File: uri/matematica/1161.py
fat = [1]

def factorial(n):
  if len(fat) <= n:
    for i in range(len(fat), n+1):
      fat.append(fat[i-1]*i)
  return fat[n]

while True:
  try:
    fat1, fat2 = map(int, input().split(' '))
    print(factorial(fat1)+factorial(fat2))  
  except EOFError:
    break

File: uri/matematica/1161.c
#include <stdio.h>
#include <stdlib.h>

int tam = 1;

long long int factorial(int n, long long int arr[]) {
  if (tam <= n) {
    int i;
    for (i = tam; i <= n; i++) {
      arr[i] = arr[i-1]*i;
      tam++;
    }
  }
  return arr[n];
}

int main () {
  int fat1, fat2;
  long long int arr[20] = {1};
  while (scanf("%d %d", &fat1, &fat2) != EOF) {
    printf("%lld\n", factorial(fat1, arr) + factorial(fat2, arr));
  }
  return 0;
}